swig/cpp/src/Internal.hpp

```diff
@@ -0,0 +1,222 @@
+    {                                                                                                                                                \
+        auto s_vector = new vector<S_String>;                                                                                                        \
+        if (NULL == s_vector) {                                                                                                                      \
+            return NULL;                                                                                                                             \
+        }                                                                                                                                            \
+                                                                                                                                                     \
+        for (uint8_t i = 0; i < data->size; i++) {                                                                                                   \
+            s_vector->push_back(std::string(data->element[i]));                                                                                      \
+        }                                                                                                                                            \
+                                                                                                                                                     \
+        return s_vector;                                                                                                                             \
+    };
+
+#include <iostream>
+#include <memory>
+
+extern "C" {
+#include "../../../src/libyang.h"
+}
+
+#define typeof(x) __typeof__(x)
+
+using namespace std;
+
+/* defined */
+class Deleter;
+
+/* used */
+class Context;
+
+typedef enum free_type_e {
+    CONTEXT,
+    DATA_NODE,
+    //TODO DATA_NODE_WITHSIBLINGS,
+    SCHEMA_NODE,
+    MODULE,
+    SUBMODULE,
+    XML,
+    SET,
+    DIFFLIST,
+} free_type_t;
+
+typedef union value_e {
+    struct ly_ctx *ctx;
+    struct lyd_node *data;
+    struct lys_node *schema;
+    struct lys_module *module;
+    struct lys_submodule *submodule;
+    struct lyxml_elem *elem;
+    struct ly_set *set;
+    struct lyd_difflist *diff;
+} value_t;
+
+class Deleter
+{
+public:
+    Deleter(ly_ctx *ctx, S_Deleter parent = NULL);
+    Deleter(struct lyd_node *data, S_Deleter parent = NULL);
+    Deleter(struct lys_node *schema, S_Deleter parent = NULL);
+    Deleter(struct lys_module *module, S_Deleter parent = NULL);
+    Deleter(struct lys_submodule *submodule, S_Deleter parent = NULL);
+    Deleter(S_Context context, struct lyxml_elem *elem, S_Deleter parent = NULL);
+    Deleter(struct ly_set *set, S_Deleter parent = NULL);
+    Deleter(struct lyd_difflist *diff, S_Deleter parent = NULL);
+    ~Deleter();
+
+private:
+    S_Context _context;
```

Comment:
nitpicking: Identifiers starting with underscore are reserved; my suggestion is to remove these.

Reply:
ok.